plugins/flutter_aepcore/lib/flutter_aepcore.dart

```diff
@@ -23,6 +23,21 @@ class MobileCore {
   static Future<String> get extensionVersion =>
       _channel.invokeMethod<String>('extensionVersion').then((value) => value!);
 
+ /// Initializes the SDK with either initialization options or an App ID. 
+ /// You must provide either `initOptions` or `appId`, but not both.
+ /// @param initOptions The [InitOptions] to configure the SDK.
+ /// @param appId The AEP SDK App ID.
+ /// @throws [ArgumentError] if neither `initOptions` nor `appId` is provided, or if both
+  static Future<void> initialize({InitOptions? initOptions, String? appId}) {
```

Comment:
Split it into two APIs:
initialize()
initializeWithAppId()
For initializeWithAppId(), create an InitOptions object using the provided appId, and then call the initialize() API.